modules/llm-cache/ds/kv_state_cache_manager.cc

```diff
@@ -82,43 +83,134 @@ Status KVStateCacheManager::Make(std::shared_ptr<KVStateCacheManager>& manager,
+ * data, and the length is the size of the tensor data.
+ * @param updated It's a return value to indicate the number of tokens that have
+ * been updated successfully.
+ *
+ * @note The length of the token list should be as same as the length of the
+ * kvStateList.
+ *
+ * @return Status
+ */
 Status KVStateCacheManager::Update(
     const std::vector<int>& prefix, const std::vector<int>& tokenList,
-    const std::vector<std::map<int, std::pair<LLMKV, LLMKV>>>& kvStateList) {
+    const std::vector<std::vector<std::pair<LLMKV, LLMKV>>>& kvStateList,
+    size_t& updated) {
   if (kvStateList.size() != tokenList.size()) {
     return Status::Invalid("Token list size not match kv state list size");
   }
-  return storage->Update(prefix, tokenList, kvStateList);
+  return storage->Update(prefix, tokenList, kvStateList, updated);
 }
 
+/**
+ * @brief Update the kv state with the given token in the kv state cache
+ * manager.
+ *
+ * @param tokenList The token list as the prefix of the next token.
+ * @param nextToken The next token to be updated.
+ * @param kvState The kv state of the next token.
+ *
+ * @return Status
+ */
 Status KVStateCacheManager::Update(
     const std::vector<int>& tokenList, int nextToken,
-    const std::map<int, std::pair<LLMKV, LLMKV>>& kvState) {
+    const std::vector<std::pair<LLMKV, LLMKV>>& kvState) {
   return storage->Update(tokenList, nextToken, kvState);
 }
 
+/**
+ * @brief Query the kv state with the given token and its prefix in the kv state
+ * cache manager.
+ *
+ * @param tokenList The token list as the prefix of the next token.
+ * @param nextToken The next token to be queried.
+ * @param kvState The kv state of the next token. It must be initialized before
+ *                calling this function, including the data and length of the kv
+ *                tensor. Also, the length of the kvState should be as same as
+ *                the layer of the kv state.
+ *
+ * @return Status
+ */
 Status KVStateCacheManager::Query(
     const std::vector<int>& tokenList, int nextToken,
-    std::map<int, std::pair<LLMKV, LLMKV>>& kvState) {
+    std::vector<std::pair<LLMKV, LLMKV>>& kvState) {
   return storage->Query(tokenList, nextToken, kvState);
 }
 
+/**
+ * @brief Query the kv state with the given token list in the kv state cache
+ * manager.
+ *
+ * @param tokenList The token list to be queried.
+ * @param kvStateList The kv state list of the token list.
+ *                    It must be initialized before calling this function,
+ * including the data and length of the kv tensor. It's a 2D vector, the first
+ * dimension is the token index, and the second dimension is the layer index.
+ *                    The kv state is a pair of LLMKV, the first is the K tensor
+ *                    and the second is the V tensor. It contains two fields:
+ *                    data and length. The data is the pointer to the tensor
+ * data, and the length is the size of the tensor data.
```

Comment:
alignment.

Reply:
Add more content about if user should pre-allocate the buffers please.

Note that blobstroage and filestoreage requires different behaviour.

Reply:
![image](https://github.com/v6d-io/v6d/assets/71587243/5021a915-6913-43b9-b603-bb0a0d4a12aa)

@sighingnow Thanks for the advice. What do you think about such a description above?